src/http.ts

```diff
@@ -141,26 +141,43 @@ enum HttpType {
   Put = "PUT",
 }
 
+type Fetch = (
+  input: RequestInfo | URL,
+  init?: RequestInit | undefined
+) => Promise<Response>;
+
 /**
  * Helps build lemmy HTTP requests.
  */
 export class LemmyHttp {
   #apiUrl: string;
   #headers: { [key: string]: string } = {};
   #pictrsUrl: string;
+  #fetch: Fetch;
 
   /**
    * Generates a new instance of LemmyHttp.
    * @param baseUrl the base url, without the vX version: https://lemmy.ml -> goes to https://lemmy.ml/api/vX
    * @param headers optional headers. Should contain `x-real-ip` and `x-forwarded-for` .
+   * @param fetch optional fetch function to use.
    */
-  constructor(baseUrl: string, headers?: { [key: string]: string }) {
+  constructor(
+    baseUrl: string,
+    headers?: { [key: string]: string },
+    fetch?: Fetch
```

Comment:
You can get rid of the `Fetch` type and just use `typeof fetch` instead. Even if the client is used in a place without a global fetch, the Typescript types won't be affected.